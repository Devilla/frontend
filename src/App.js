import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
