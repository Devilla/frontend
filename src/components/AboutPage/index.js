import React, { Component } from 'react';
import {Animated} from "react-animated-css";
export default class About extends Component {
     componentDidMount(){
     window.scrollTo(0, 0)
   } 
  render() {
    return (
        <div>
            <div className="page-header"><div className="page-header-overlay">
            <div className="centered page-header-container w-container">
            <Animated className="page-header-title" animationIn="fadeInDown" animationOut="fadeOut" isVisible={true}>About us</Animated>
            </div></div></div>
            <div className="section innerpage">
                <div className="container w-container">
                    <p>We started Influence with one mission: to cut down the advertising cost and to convert more customers on your website.</p>
            <p>We use your existing customers’ data on your website and show it to your visitors so that they are socially influenced by buying behavior of people on the website.</p>
            <p>Using this tool we’ve been able to help a lot of businesses in increasing their money making and get more conversions.</p>
            <p>Social influence is the next big thing in marketing for influencing buyers to buy more.</p>
            <p>Let’s start with your free to start trial and get making more money for you.
            </p>
                   
                </div>
              </div>
        </div>
    );
  }
}
