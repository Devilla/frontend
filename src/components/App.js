import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
