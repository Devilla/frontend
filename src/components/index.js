export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
