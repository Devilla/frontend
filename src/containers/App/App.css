@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
