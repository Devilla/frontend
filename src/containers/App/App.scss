@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
