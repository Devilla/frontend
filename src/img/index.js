// Home Page Image
export { default as LogoInfluence } from './logo-influence.png';
export { default as HigherConversion } from './new-convert-more-customers.svg';
export { default as SaveAcquisitions } from './new-customer-acquisition-costs.svg';
export { default as IncreaseVisitors } from './new-review-trust.svg';
export { default as SocialInfluence } from './new-social-influence-1.svg';
export { default as Device } from './device-1.png';
export { default as AnalyticsInsights } from './user_analytics.jpg';
export { default as EasiestSetup } from './simple_setup.jpg';
export { default as ActivityNotifications } from './new-recent-purchases.svg';
export { default as LiveViewers } from './new-currently-viewing.svg';
export { default as CustomerProfiles } from './new-profiling.svg';
export { default as CustomerJourneys } from './new-customer-journey.svg';
export { default as CustomeRules } from './new-popup-timings.svg';
export { default as MobileReady } from './new-mobile-ready.svg';

export { default as Inc42 } from './inc42glyph.png';
export { default as Tnwg } from './tnwglyph3.png';
export { default as Yourstory } from './yourstoryglyph3.png';
export { default as HigherConversion2 } from './higherconversion2.svg';
export { default as Saveonacquisitions2 } from './saveonacquisitions2.svg';
export { default as Visitortrust2 } from './visitortrust2.svg';
export { default as Socialinfluence2 } from './socialinfluence2.svg';
export { default as Startconverting } from './startconverting.png';
export { default as D12 } from './D12.png';
export { default as B24 } from './B24.png';
export { default as Customerjourney5 } from './customerjourney5.png';
export { default as Analyticsinsights10 } from './analyticsinsights10.png';
export { default as Easiestsetup } from './easiestsetup.png';
export { default as Activitynotifications2 } from './activitynotifications2.svg';
export { default as Liveviewers2 } from './liveviewers2.svg';
export { default as Gdpr }  from './gdpr.png';
export { default as Integration } from './integration.png';
export { default as Swivelscreen } from './swivelscreen.png';
export { default as Sw1 }  from './sw1.png';
export { default as Sw2 }  from './sw2.png';
export { default as Marvel } from './marvel.png';
export { default as Lawsikho }  from './lawsikho.png';
export { default as Stagephod } from './stagephod.png';
export { default as Userc } from './usercircle.png';
export { default as Userr }  from './userrect.png';
export { default as Carpathy } from './carpathy.png';
export { default as Tbars } from './3bars.png';
export { default as Smallicon } from './smallicon.png';
export { default as Ticon } from './3icon.png';
export { default as Profile } from './profile.png';





// How it works Page Image
export { default as NewIntegrate } from './new-integrate.svg';
export { default as NewInstall } from './new-install.svg';
export { default as NewGoLive } from './new-go-live.svg';
export { default as NewAnalyze } from './new-analyze.svg';
export { default as Customerprofiles2 } from './customerprofiles2.svg';
export { default as Customerjourney2 } from './customerjourney2.svg';
export { default as Customrules2 } from './customrules2.svg';
export { default as Mobileready2 } from './mobileready2.svg';


//Notifications Page Image
export { default as GroupActivity } from './group activity card 9.png';
export { default as RecentActivity } from './recent activity card 6.png';
export { default as LiveActivity } from './live activity card 7.png';


// Integrations Page Image
export { default as Integrations1 } from './integrations1.png';
export { default as NewRecentPurchases } from './new-recent-purchases.svg';

// Contact Us Page
export { default as ContactAvatar1 } from './contactavatar1.png';
export { default as ContactAvatar2 } from './contactavatar2.png';
export { default as ContactAvatar3 } from './contactavatar5.png';
export { default as ContactAvatar4 } from './contactavatar7.png';
export { default as ContactAvatar5 } from './contactavatar11.png';
export { default as ContactAvatar6 } from './contactavatar15.png';

// Signup Page
export { default as SignUp } from './signuppage2.png';

//Avatars
export { default as Avatar1 } from './avatar-1.jpg'
export { default as Avatar2 } from './avatar-2.jpg'
export { default as Avatar3 } from './avatar-3.jpg'
export { default as Avatar4 } from './avatar-4.jpg'
export { default as Avatar5 } from './avatar-5.jpg'
export { default as Avatar6 } from './avatar-6.jpg'
export { default as Avatar7 } from './avatar-7.jpg'
export { default as Avatar8 } from './avatar-8.jpg'
export { default as Avatar9 } from './avatar-9.jpg'
export { default as Avatar10 } from './avatar-10.jpg'
