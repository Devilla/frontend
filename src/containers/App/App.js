import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
