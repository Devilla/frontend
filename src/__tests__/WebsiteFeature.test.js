import React from 'react'
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import renderer from 'react-test-renderer'
import { WebsiteFeature } from 'components';
import { store } from 'App';
import { Gdpr } from 'img';
import { Link } from 'react-router';

// Snapshot for website feature React Component
describe('>>>F E A T U R  E --- Snapshot',()=>{
    it('+++capturing Snapshot of feature', () => {
        const renderedValue =  renderer.create(<WebsiteFeature store={store} />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});
