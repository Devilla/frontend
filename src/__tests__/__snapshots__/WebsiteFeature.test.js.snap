// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>F E A T U R  E --- Snapshot +++capturing Snapshot of feature 1`] = `
<div
  className="WebsiteFeature-container"
>
  <div
    className="main-container"
  >
    <section
      className="cover text-center mt-5 mb-5  "
      data-overlay="0"
    >
      <div
        className="container  mt-4 mb-5 pb-0 pt-0"
      >
        <div
          className="row"
        >
          <div
            className="col-md-10 col-lg-10"
          >
            <h2>
              Power in your hands!
            </h2>
            <p
              className="lead col-md-12 text-center"
            >
               From a team of passionate creators & creative thinkers.
            </p>
            <a
              className="btn btn--primary type--uppercase"
              onClick={[Function]}
              style={Object {}}
            >
               
              <span
                className="btn__text"
              >
                Try influence now
              </span>
               
            </a>
            <div />
          </div>
        </div>
      </div>
    </section>
    <hr
      className="my-auto col-md-6"
    />
    <section
      className="imageblock switchable feature-large unpad--top unpad--bottom space--sm"
    >
      <div
        className="imageblock__content col-lg-6 col-md-4 pos-right"
      >
        <img
          alt="newimage"
          src="3bars.png"
        />
      </div>
      <div
        className="container unpad--bottom"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-md-7"
          >
            <h2>
              Customer Activity Notifications
            </h2>
            <p
              className="lead"
            >
              A stamp of 'Social Proof' & trust for your business. Designed to target customer psyche.
            </p>
          </div>
        </div>
      </div>
    </section>
    <hr
      className="my-auto col-md-6 text-right"
    />
    <section
      className="imageblock switchable switchable--switch unpad--top unpad--bottom feature-large  space--sm"
    >
      <div
        className="imageblock__content col-lg-6 col-md-4 pos-right"
      >
        <img
          alt="newimage"
          src="smallicon.png"
        />
      </div>
      <div
        className="container unpad--bottom"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-md-7"
          >
            <h2>
              Plug & Play
            </h2>
            <p
              className="lead"
            >
              Carefully set configurations, start in less than 3 minutes.
            </p>
          </div>
        </div>
      </div>
    </section>
    <hr
      className="my-auto col-md-6"
    />
    <section
      className="imageblock switchable feature-large unpad--top unpad--bottom space--sm"
    >
      <div
        className="imageblock__content col-lg-6 col-md-4 pos-right"
      >
        <img
          alt="newimage"
          src="3icon.png"
        />
      </div>
      <div
        className="container unpad--bottom"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-md-7"
          >
            <h2>
              Customize Your Own
            </h2>
            <p
              className="lead"
            >
              Everything is customizable! From notification designs to their behaviour.
            </p>
          </div>
        </div>
        =
      </div>
    </section>
    <hr
      className="my-auto col-md-6 text-right"
    />
    <section
      className="imageblock switchable switchable--switch  unpad--top unpad--bottom feature-large space--sm"
    >
      <div
        className="imageblock__content col-lg-6 col-md-4 pos-right"
      >
        <img
          alt="newimage"
          className="profile"
          src="3profile.png"
        />
      </div>
      <div
        className="container unpad--bottom"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-md-7"
          >
            <h2>
              Customer Identity & Analytics
            </h2>
            <p
              className="lead"
            >
              Know the true identity of your customer. Analyse your customer's path & behaviour on your web pages.
            </p>
          </div>
        </div>
      </div>
    </section>
    <hr
      className="my-auto col-md-6 text-right"
    />
  </div>
</div>
`;
